satpy/readers/modis_l3.py

```diff
@@ -0,0 +1,132 @@
+        pos2 = gridname.rfind("Deg")
+
+        # Some products don't have resolution listed.
+        if pos < 0 or pos2 < 0:
+            self.resolution = 360. / self.ncols
+        else:
+            self.resolution = float(gridname[pos:pos2])
+
+    def _sort_grid(self):
+        """Get the grid properties."""
+        # First, get the grid resolution
+        self._get_res()
+
+        # Now compute the data extent
+        upperleft = self.metadata["GridStructure"]["GRID_1"]["UpperLeftPointMtrs"]
+        lowerright = self.metadata["GridStructure"]["GRID_1"]["LowerRightMtrs"]
+
+        # For some reason, a few of the CMG products multiply their
+        # decimal degree extents by one million. This fixes it.
+        if lowerright[0] > 1e6:
+            upperleft = tuple(val / 1e6 for val in upperleft)
+            lowerright = tuple(val / 1e6 for val in lowerright)
+
+        return (upperleft[0], lowerright[1], lowerright[0], upperleft[1])
+
+
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Init the file handler."""
+        super().__init__(filename, filename_info, filetype_info, **kwargs)
+
+        # Initialise number of rows and columns
+        self.nrows = self.metadata["GridStructure"]["GRID_1"]["YDim"]
+        self.ncols = self.metadata["GridStructure"]["GRID_1"]["XDim"]
+
+        # Get the grid name and other projection info
+        self.area_extent = self._sort_grid()
+
+
+    def available_datasets(self, configured_datasets=None):
+        """Automatically determine datasets provided by this file."""
+        logger.debug("Available_datasets begin...")
+
+        ds_dict = self.sd.datasets()
+
+        yield from super().available_datasets(configured_datasets)
+        common = {"file_type": "mcd43_cmg_hdf", "resolution": self.resolution}
+        for key in ds_dict.keys():
+            if "/" in key:  # not a dataset
+                continue
+            yield True, {"name": key} | common
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get DataArray for specified dataset."""
+        dataset_name = dataset_id["name"]
+        dataset = self.load_dataset(dataset_name, dataset_info.pop("category", False))
+        self._add_satpy_metadata(dataset_id, dataset)
+
+        return dataset
+
+
+    def get_area_def(self, dsid):
+        """Get the area definition.
+
+        This is fixed, but not defined in the file. So we must
+        generate it ourselves with some assumptions.
+        """
+        proj_param = "EPSG:4326"
+
+        area = geometry.AreaDefinition("gridded_modis",
+                                       "A gridded L3 MODIS area",
+                                       "longlat",
+                                       proj_param,
+                                       self.ncols,
+                                       self.nrows,
+                                       self.area_extent)
+        self.area = area
+
+        return self.area
```

Comment:
```suggestion
                                       self.area_extent)
        return area
```

Reply:
I'd also really prefer if `self.area_extent` didn't have to be an instance attribute. If the main reason is so we can have the error from `_get_res` about only CMG grids being supported and have it raised on file handler creation, then fine. But otherwise, doing those checks on `__init__` is only slowing down Scene creation (admittedly probably not much) when datasets may not be loaded from this file.

Reply:
Aah, I asked about this on slack (https://pytroll.slack.com/archives/C0LNH7LMB/p1699437836789399) but misunderstood your reply. I'll remove the code.